### PR TITLE
chore: Enable flutter analyze check

### PR DESCRIPTION
**Pull request by drminh2807**

### CODE REVIEW

**Comment by nqhhdev** [comment]
Why do we need?

**Comment by sherlockvn** [comment]
> _ No description provided. _

Be careful to use it, because some place need to ignore linter to work as expected

File: scripts/code_analyze.sh [hunk 1]
```diff
@@ -3,5 +3,6 @@ flutter pub get && flutter pub run build_runner build --delete-conflicting-outpu
 flutter pub run import_sorter:main --no-comments
 flutter format lib/ test/
 flutter pub get
-flutter analyze --no-fatal-infos --no-fatal-warnings
+flutter analyze
```
Comment (dab246) [inline scripts/code_analyze.sh:6]: IMO, `flutter analyze --no-fatal-infos --no-fatal-warnings` should be left as is. 
Treating `infos` or `warnings` as `fatal` is too strict at a time when the project is still in its early stages.

**Review by dab246 — COMMENTED** [review]
Suggest using [`GitHub Action — Analyze Dart`](https://github.com/marketplace/actions/analyze-dart) to analyze the project.

**Comment by hoangdat** [comment]
waiting for `super-thursday`: 3 Aug 23

**Comment by tk-nguyen** [comment]
FYI, the CI is running analyze on every PR, so in a way we're already using code analysis

**Pull request closed without merging**